pandapower/plotting/plotly/traces.py

```diff
@@ -480,14 +480,15 @@ def _create_branch_trace(net, branches=None, use_branch_geodata=True, respect_se
         else:
             raise NotImplementedError("respect separtors is only implements for switches, "
                                       "not for {}s.".format(separator_element))
-    branches_to_plot = net[branch_element].loc[set(net[branch_element].index) & (set(branches) - no_go_branches)]
+    branches_to_plot = net[branch_element].loc[list(set(net[branch_element].index) &
```

Comment:
I suggest using np.intersect1d: it works for arrays like & works for sets and it returns an array directly

Reply:
I timed it, and Python set operations are way faster than numpy set operations. BUT: casting arrays to set, then performing Python set operations, and casting the result to list is slower than numpy set operations.

Reply:
I suggest using Python set operations when the variables are sets already, and otherwise to use numpy set operations

Reply:
numpy.intersect1d is difficult here, because a pd.index, a list and a set are combined